### PR TITLE
Improve diagnostics of hlint ideas

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR improves conversion of hlint ideas to file diagnostics. Particularly, the src span of the idea is now recorded.

### CODE REVIEW

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs [hunk 1]
```diff
@@ -543,17 +544,26 @@ getHlintDiagnosticsRule =
...
+              , _character  = HSE.srcSpanStartColumn span }
+        , _end   = LSP.Position {
+                _line = HSE.srcSpanEndLine span
+             , _character = HSE.srcSpanEndColumn span }
+        }
+      diagnostic :: NormalizedFilePath -> Idea -> FileDiagnostic
+      diagnostic file i =
+        (file, LSP.Diagnostic {
+              _range = srcSpanToRange $ ideaSpan i
+            , _severity = Just LSP.DsInfo
```
Comment (cocreature) [inline compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: I wonder if `Info` is the correct priority here or if we should make it a warning. Probably a good idea to check what linters for other languages do. (But no need to address this as part of this PR).

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs [hunk 2]
```diff
@@ -543,17 +544,26 @@ getHlintDiagnosticsRule =
...
-         _range = noRange,
-         _severity = Just LSP.DsInfo,
-         _code = Nothing,
-         _source = Just "linter",
-         _message = msg,
-         _relatedInformation = Nothing
+      srcSpanToRange :: HSE.SrcSpan -> LSP.Range
+      srcSpanToRange span = Range {
+          _start = LSP.Position {
+                _line = HSE.srcSpanStartLine span
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: Thank-you for point this out. Will fix!

File: compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs [hunk 3]
```diff
@@ -543,17 +544,26 @@ getHlintDiagnosticsRule =
...
-      -- To-do : Improve this.
-      toDiagnostic file i = ideHintText file (T.pack $ show i)
-      ideHintText fp msg = (fp, LSP.Diagnostic {
-         _range = noRange,
-         _severity = Just LSP.DsInfo,
-         _code = Nothing,
-         _source = Just "linter",
-         _message = msg,
-         _relatedInformation = Nothing
+      srcSpanToRange :: HSE.SrcSpan -> LSP.Range
```
Comment (hurryabit) [inline compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: I would guess we already have this function somewhere else. I don't know where though.

Comment (cocreature) [reply compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs]: We don’t. There are multiple things called `SrcSpan`. This is the one from `haskell-src-exts` which is new.

**Pull request merged**